### PR TITLE
Add check for NFF access key

### PR DESCRIPTION
**Pull request by leonardoam**
Problem:

- There are new rules for NFe access key validation under NFF regime

Solution:

- Add these new validations on a new check method called CheckNFF on the
  accesskey package

### CODE REVIEW

File: sefaz/accesskey/validation.go [hunk 1]
```diff
@@ -18,7 +18,27 @@ func Check(accessKey AccessKey) error {
 	return nil
 }
 
-// validate execute all sub-routines necessary to perform a full accesskey validation
+// CheckNFF checks if an access key is valid also considering NFF rules. It returns an error with an specific code based on
+// the validation proble
+func CheckNFF(accessKey AccessKey) error {
```
Comment (andremissaglia) [inline sefaz/accesskey/validation.go:23]: Shouldn't this validation extend the already-existing `Check` function? It seems strange to me that the user is forced to choose between `Check` and `CheckNFF`

File: sefaz/accesskey/validation_test.go [hunk 2]
```diff
@@ -13,9 +13,15 @@ func TestCheck_ShouldReturnNilForValidAccessKeys(t *testing.T) {
 	err := Check(key)
 
 	assert.Nil(t, err)
+
+	key = AccessKey("35210501180169000133551210501100013022024517")
```
Comment (andremissaglia) [inline sefaz/accesskey/validation_test.go:17]: Nit: I think it would be better to have more than one success case in the unit test

**Review by andremissaglia — COMMENTED** [review]
My only concern here is that there are two separate functions for validations. Not only this will force a refactor in a lot of projects, but also I can't imagine someone handling only one type of accesskey

**Comment by eric-reis** [comment]
> My only concern here is that there are two separate functions for validation. Not only this will force a refactor in a lot of projects, but also I can't imagine someone handling only one type of accesskey

All NFF Access Keys are valid under the current Access Key validation, being NFF a more restrictive approach. It would be slightly different if current access key validation checked for emission type - which it does not because there's no clear date in which the field started to exist in the access key.

Thus. there's no refactor needed in any system as long as they don't need to explicitly detect if a valid access key is also a valid NFF access key. On the flip side, if a "complete" access key validation is to be done at once, in a single method, a discovery would be needed as a correct solution has rules dependent on time of change of rules, including transition interval in which two rulesets are active.

**Review by andremissaglia — APPROVED** [review]

File: sefaz/accesskey/validation.go [hunk 3]
```diff
@@ -18,7 +18,27 @@ func Check(accessKey AccessKey) error {
...
+	}
+
+	if isNFF(accessKey) {
+		err := validateNFF(accessKey)
+		if err != nil {
+			return errors.E(op, err)
+		}
+	}
+
+	return nil
```
[the remark this answers is not in the document]
Comment (leonardoam) [reply sefaz/accesskey/validation.go:38]: I couldn't find a better name, but this method will check for the sames rules as `Check(accesskey AccessKey)` but considering additional rules for NFF. It's explained in the docs.

**Comment by leonardoam** [comment]
Just to add something to what you already discussed about Check vs CheckNFF: I think it probably will make sense to refactor it in the future to be the same method. We don't have enough NFF samples so it's hard to know right now if some rules will in fact apply, and there are other checks that I would apply but I'm not sure that Sefaz will follow, so I just used the ones described in the NT. The trade-off is maybe causing a non retro-compatible refactor in the future.

Finally, I added a little more information in the docs about CheckNFF being more restrictive.

Other suggestions will be fixed on my next commit.

**Review by eric-reis — APPROVED** [review]

**Pull request merged**